version: '3.8'

services:
  traefik:
    image: traefik:v2.11
    container_name: traefik
    command:
      - "--api.dashboard=true"
      - "--api.insecure=true"  
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
      - "--entrypoints.web.address=:80"
    ports:
      - "80:80"
      - "8080:8080"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
    networks:
      - traefik

  gateway:
    build: .
    container_name: flask-gateway
    networks:
      - traefik
    ports:
      - "5004:5004"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.gateway.rule=PathPrefix(`/`)"
      - "traefik.http.services.gateway.loadbalancer.server.port=5004"

  user-service:
    image: ../user-service/
    container_name: user-service
    networks:
      - traefik
    ports:
      - "5004:5004"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.api1.rule=PathPrefix(`/login`)"
      - "traefik.http.routers.api1.entrypoints=web"
      - "traefik.http.services.api1.loadbalancer.server.port=5001"

networks:
  traefik:
    external: false
